lib/rucio/rse/protocols/webdav.py

```diff
@@ -89,56 +90,63 @@ def __len__(self):
         return self.length
 
 
-class Parser:
+@dataclass(frozen=True)
+class _PropfindFile:
+    """Contains the properties of one file from a PROPFIND response."""
 
-    """ Parser to parse XML output for PROPFIND ."""
+    href: Optional[str]
+    size: Optional[int]
```

Comment:
Is there a use case where we recieve one of `href`, `size`, but not the other?

From my perspective these are tighly coulped, I'd rather see a `None` there opposed to `_PropfindFile(None, 22)` or `_PropfindFile(path, None)`, which both does not make sense?

At least `href` should not be Optional...

Reply:
It's just because I can't guarantee what the WebDAV server responds with. I agree that `href` could be made mandatory (it is in the RFC at least), so I'll change it to throw an exception if it is missing from the response.